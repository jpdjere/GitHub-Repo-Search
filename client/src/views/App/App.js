import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  componentDidMount(){
    console.log("If proxy works, should return users object. Remember to start backend server on localhost:3001:");
    fetch('/users')
     .then(res => res.json())
     .then(users => console.log({ users }));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
