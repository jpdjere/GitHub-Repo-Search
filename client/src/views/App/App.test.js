import React from 'react';
import ReactDOM from 'react-dom';
import {App} from './App';

import renderer from 'react-test-renderer';

describe('App', () => {

 it('should be defined', () => {
   expect(App).toBeDefined();
 });

 it('should match the snapshot', () => {
   // const tree = renderer.create(<App />).toJSON();
   const tree = shallow(<App />);
   expect(tree).toMatchSnapshot();
 });

});

it('sums numbers', () => {
  let sum = (a,b) => a + b;
  expect(sum(1, 2)).toEqual(3);
  expect(sum(2, 2)).toEqual(4);
});
