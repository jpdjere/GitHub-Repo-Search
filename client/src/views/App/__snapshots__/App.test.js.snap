// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <HashRouter>
        <div>
                <Route
                        component={[Function]}
                        path="/"
                />
                <div
                        className="search__searchBarContainer"
                >
                        <withRouter(Connect(ReduxForm)) />
                </div>
                <Switch>
                        <Route
                                component={[Function]}
                                exact={true}
                                name="Search"
                                path="/"
                        />
                        <Route
                                component={[Function]}
                                exact={true}
                                name="Top Contributors"
                                path="/top/:owner/:repo"
                        />
                </Switch>
        </div>
</HashRouter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Route
                    component={[Function]}
                    path="/"
          />
          <div
                    className="search__searchBarContainer"
          >
                    <withRouter(Connect(ReduxForm)) />
          </div>
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              name="Search"
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              name="Top Contributors"
                              path="/top/:owner/:repo"
                    />
          </Switch>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/"
/>,
            <div
              className="search__searchBarContainer"
>
              <withRouter(Connect(ReduxForm)) />
</div>,
            <Switch>
              <Route
                            component={[Function]}
                            exact={true}
                            name="Search"
                            path="/"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            name="Top Contributors"
                            path="/top/:owner/:repo"
              />
</Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <withRouter(Connect(ReduxForm)) />,
              "className": "search__searchBarContainer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  name="Search"
                  path="/"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  name="Top Contributors"
                  path="/top/:owner/:repo"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "name": "Search",
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "name": "Top Contributors",
                  "path": "/top/:owner/:repo",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <HashRouter>
          <div>
                    <Route
                              component={[Function]}
                              path="/"
                    />
                    <div
                              className="search__searchBarContainer"
                    >
                              <withRouter(Connect(ReduxForm)) />
                    </div>
                    <Switch>
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        name="Search"
                                        path="/"
                              />
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        name="Top Contributors"
                                        path="/top/:owner/:repo"
                              />
                    </Switch>
          </div>
</HashRouter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <Route
                        component={[Function]}
                        path="/"
            />
            <div
                        className="search__searchBarContainer"
            >
                        <withRouter(Connect(ReduxForm)) />
            </div>
            <Switch>
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    name="Search"
                                    path="/"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    name="Top Contributors"
                                    path="/top/:owner/:repo"
                        />
            </Switch>
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/"
/>,
              <div
                className="search__searchBarContainer"
>
                <withRouter(Connect(ReduxForm)) />
</div>,
              <Switch>
                <Route
                                component={[Function]}
                                exact={true}
                                name="Search"
                                path="/"
                />
                <Route
                                component={[Function]}
                                exact={true}
                                name="Top Contributors"
                                path="/top/:owner/:repo"
                />
</Switch>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <withRouter(Connect(ReduxForm)) />,
                "className": "search__searchBarContainer",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    name="Search"
                    path="/"
/>,
                  <Route
                    component={[Function]}
                    exact={true}
                    name="Top Contributors"
                    path="/top/:owner/:repo"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "name": "Search",
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "name": "Top Contributors",
                    "path": "/top/:owner/:repo",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
